src/full/Agda/Interaction/Highlighting/Common.hs

```diff
@@ -0,0 +1,57 @@
+-- | Common syntax highlighting functions for Emacs and JSON
+
+module Agda.Interaction.Highlighting.Common
+  ( toAtoms
+  , chooseHighlightingMethod
+  , writeToTempFile
+  ) where
+
+import Agda.Interaction.Highlighting.Precise
+import Agda.Syntax.Common
+import Agda.TypeChecking.Monad (HighlightingMethod(..))
+import qualified Agda.Utils.IO.UTF8 as UTF8
+
+import qualified Control.Exception as E
+import Data.Maybe (maybeToList)
+import Data.Char (toLower)
+import qualified System.Directory as D
+import qualified System.IO as IO
+
+-- | Converts the 'aspect' and 'otherAspects' fields to strings that are
+-- friendly to editors.
+toAtoms :: Aspects -> [String]
+toAtoms m = map toAtom (otherAspects m) ++ toAtoms' (aspect m)
+  where
+  toAtom :: Show a => a -> String
+  toAtom = map toLower . show
+
+  kindToAtom (Constructor Inductive)   = "inductiveconstructor"
+  kindToAtom (Constructor CoInductive) = "coinductiveconstructor"
+  kindToAtom k                         = toAtom k
+
+  toAtoms' Nothing               = []
+  toAtoms' (Just (Name mKind op)) =
+    map kindToAtom (maybeToList mKind) ++ opAtom
+    where opAtom | op        = ["operator"]
+                 | otherwise = []
+  toAtoms' (Just a) = [toAtom a]
+
+-- | Creates a temporary file, writes some stuff, and returns the filepath
+writeToTempFile :: String -> IO FilePath
+writeToTempFile content = do
```

Comment:
This should probably go in the `Agda/Utils` directory to make
it easier for others to notice the function and reuse it.

Given that `openTempFile` is in `System.IO`, `Agda.Utils.IO`
could be a good candidate. Or maybe a new `Agda.Utils.TempFile`?

Reply:
I think `Agda.Utils.TempFile` is a good idea.